### PR TITLE
Refactoring of Plugin examples to match Radium V1 build chain

### PR DESCRIPTION
**Pull request by MathiasPaulin**
Plugins can be compiled and installed into the Radium Bundle to be used by all applications. If an application that uses a plugin need to be moved on another directory (or computer), the application MUST be installed AFTER the plugins.
 
A general documentation about Radium installation process, Application development and Plugin development is still missing.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,43 +1,5 @@
-cmake_minimum_required(VERSION 2.8.11)
-project(Radium-PluginExamples)
+cmake_minimum_required(VERSION 3.6)
+project(Radium-apps)
```
Comment (hoshiryu) [inline CMakeLists.txt]: Should stay as PluginExamples

File: fake-toon/README.md [hunk 2]
```diff
@@ -0,0 +1 @@
+# Fake Toon Radium plugin example
```
Comment (hoshiryu) [inline fake-toon/README.md:1]: Tell me more

File: laplacian-smoothing/README.md [hunk 3]
```diff
@@ -0,0 +1 @@
+# Laplacian smoothing Radium plugin example
```
Comment (hoshiryu) [inline laplacian-smoothing/README.md:1]: Tell me more

**Review by hoshiryu — CHANGES_REQUESTED** [review]
Regarding Plugins creating a UI widget, it is more secure to create it when requested.
Why not using the `_ra` suffix?
It is not radium, but would it be interesting to use the same formatting?

File: laplacian-smoothing/README.md (hunk 3 above)
Comment (MathiasPaulin) [reply laplacian-smoothing/README.md]: > Regarding Plugins creating a UI widget, it is more secure to create it when requested.
This is out of scope of this PR. The old plugin example code was not modified nor refactored except to be consistant with the Radium version of PR #462. 
> Why not using the `_ra` suffix?
cf above
> It is not radium, but would it be interesting to use the same formatting?
cf above

Comment (MathiasPaulin) [reply laplacian-smoothing/README.md:1]: This is a todo

**Comment by MathiasPaulin** [comment]
> Regarding Plugins creating a UI widget, it is more secure to create it when requested.

This is out of scope of this PR. The old plugin example code was not modified nor refactored except to be consistant with the Radium version of PR #462.

> Why not using the `_ra` suffix?

cf above

> It is not radium, but would it be interesting to use the same formatting?

cf above

**Comment by hoshiryu** [comment]
I have the following directory structure:
```
Radium/
    ├── Radium-Apps/
    ├── Radium-Engine/
    └── Radium-PluginExample/
```
With Radium installed in `Radium-Engine/install_radium`.
I could configure, compile install (`Radium-Apps/install_apps`) and run the apps, but I cannot configure the Plugins from `Radium-PluginExample/build` (following the `README.md` file):
```
$> cmake -DCMAKE_MODULE_PATH=../../Radium-Engine/cmake -DCMAKE_BUILD_TYPE=Release ..
-- The C compiler identification is GNU 8.3.0
-- The CXX compiler identification is GNU 8.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at fake-toon/CMakeLists.txt:14 (find_package):
  By not providing "FindRadium.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Radium", but
  CMake did not find one.

  Could not find a package configuration file provided by "Radium" with any
  of the following names:

    RadiumConfig.cmake
    radium-config.cmake

  Add the installation prefix of "Radium" to CMAKE_PREFIX_PATH or set
  "Radium_DIR" to a directory containing one of the above files.  If "Radium"
  provides a separate development package or SDK, be sure it has been
  installed.


-- Configuring incomplete, errors occurred!
```
I couldn't find the file `RadiumConfig.cmake` in `Radium-Engine/cmake`, nor `Radium-Engine/install_radium`

Thus I tried using `CMAKE_PREFIX_PATH` instead of `CMAKE_MODULE_PATH` (just like is done for the apps), which gave me:
```
$ cmake .. -DCMAKE_BUILD_TYPE=Release -DCMAKE_PREFIX_PATH=../../Radium-Engine/install_radium
-- The C compiler identification is GNU 8.3.0
-- The CXX compiler identification is GNU 8.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
Set install prefix to /ssd/canezin/Radium_cmake/Radium-PluginExample/build/fake-toon/installed-GNU
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Found OpenMP_C: -fopenmp (found version "4.5") 
-- Found OpenMP_CXX: -fopenmp (found version "4.5") 
-- Found OpenMP: TRUE (found version "4.5")  
CMake Error at fake-toon/CMakeLists.txt:63 (configure_radium_plugin_install):
  Unknown CMake command "configure_radium_plugin_install".


-- Configuring incomplete, errors occurred!
See also "/ssd/canezin/Radium_cmake/Radium-PluginExample/build/CMakeFiles/CMakeOutput.log".
See also "/ssd/canezin/Radium_cmake/Radium-PluginExample/build/CMakeFiles/CMakeError.log".
```

**Review by hoshiryu — APPROVED** [review]

**Review by hoshiryu — CHANGES_REQUESTED** [review]
The main README.md file should be updated w.r.t. the cmake configuration steps (which ressembles the one from the Radium-Apps)!

**Pull request merged**